ENVIRONMENT.md

```diff
@@ -0,0 +1,67 @@
+# Environment for building etc
+
+This repository contains many scripts for builds, releases, continuous
+integration, etc. This file describes the environment prerequisites
+for those scripts.
+
+## Handy Dockerfile for working on Linux
+
+Installing some of the dependencies can be tricky, particularly on
+restricted internal versions of Linux. For that reason, we provide a
+[Dockerfile](docker/Dockerfile) to get started with. The typical
+procedure would be:
+
+- `git clone https://github.com/googleapis/google-cloud-dotnet.git`
+- `cd google-cloud-dotnet`
+- `docker build -t google-cloud-dotnet docker`
```

Comment:
Is that final `docker` meant to be there?

Reply:
Yes. That's where the Dockerfile lives, and we don't want the Docker image to contain any of the rest of the local context.